Framework/Algorithms/inc/MantidAlgorithms/FindPeakBackground.h

```diff
@@ -65,6 +67,57 @@ class DLLExport FindPeakBackground : public API::Algorithm {
                           const size_t p_min, const size_t p_max,
                           const bool hasPeak, double &out_bg0, double &out_bg1,
                           double &out_bg2);
+
+  /// process inputs
+  void processInputProperties();
+
+  /// create output workspace
+  void createOutputWorkspaces();
+
+  /// main method to find peak background
+  int findBackground(std::vector<size_t> &peak_min_max_indexes,
+                     std::vector<double> &bkgd3);
+
+  /// set histogram data to find background
+  void setHistogram(const HistogramData::Histogram &histogram);
+
+  /// set sigma constant
+  void setSigma(const double &sigma);
+
+  /// set background order
+  void setBackgroundOrder(size_t order);
+
+  /// set fit window
+  void setFitWindow(const std::vector<double> &window);
+
+  /// add result to output workpsace
+  void addResultToOutput(const int &goodfit,
+                         const std::vector<size_t> &peakindexes,
+                         const std::vector<double> &bkgdparams);
+
+  /// histogram data to find peak background
+  boost::shared_ptr<const HistogramData::Histogram> m_histogram;
+
+  //  HistogramData::Histogram& m_histogram;
+  /// fit window
+  std::vector<double> m_vecFitWindows;
+  size_t m_iStartX; /// l0 in original code
+  size_t m_iStopX;  /// n in original code
+  /// background order: 0 for flat, 1 for linear, 2 for quadratic
+  size_t m_backgroundOrder;
+  /// constant sigma
+  double m_sigmaConstant;
+  /// output workspace (table of result)
+  API::ITableWorkspace_sptr m_outPeakTableWS;
+
+  size_t m_inputWSIndex;
```

Comment:
(minor)  This order seems odd to me. What would benefit from being in the first cache line?